dali/pipeline/operator/builtin/input_operator.h

```diff
@@ -328,6 +328,20 @@ class InputOperator : public Operator<Backend>, virtual public BatchSizeProvider
   }
 
 
+  /**
+   * "depleted" operator trace specifies whether the operator has sufficient recourses to
+   * be run in yet another iteration.
+   *
+   * If "false", the operator needs to be fed with data to run the next iteration. If "true",
+   * the next iteration can be triggered.
+   * @param ws Current workspace.
+   * @param depleted Value of the trace.
+   */
+  virtual void SetDepletedOperatorTrace(Workspace& ws, bool depleted) {
+    ws.SetOperatorTrace("depleted", depleted ? "true" : "false");
+  }
```

Comment:
Since this only sets an operator trace, does it need to be `virtual`? It seems that whatever complex logic may be there to determine whether an input operator is depleted or not, it's before the setting the trace.

Reply:
Removed the `virtual`